FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Exceptions/ErrorProcessors/JobExecutionErrorProcessor.cs

```diff
@@ -0,0 +1,57 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Microsoft.Health.Fhir.Synapse.Common.Exceptions;
+using Microsoft.Health.Fhir.Synapse.Common.Metrics;
+using Microsoft.Health.Fhir.Synapse.DataClient.Exceptions;
+using Microsoft.Health.Fhir.Synapse.DataWriter.Exceptions;
+using Microsoft.Health.Fhir.Synapse.SchemaManagement.Exceptions;
+
+namespace Microsoft.Health.Fhir.Synapse.Core.Exceptions.ErrorProcessors
+{
+    public class JobExecutionErrorProcessor : IJobExecutionErrorProcessor
```

Comment:
Just doubt whether we need these error processors just wrapping the IMetricsLogger? Maybe we can add a extension to MetricsLogger like `LogExceptionMetrics(Exception ex, OperationType operationType, string message)` Then error type can be infered from ex and operationType in some OperationErrorHandler. Thus in the code we only see logger, metricslogger, diagnosticlogger

Reply:
Updated it, thanks.